### PR TITLE
Sign Extender

### PR DESCRIPTION
**Pull request by MMaluschka13**
Matthew: Wrote sign extender.

### CODE REVIEW

File: python/Sign_Extender.py [hunk 1]
```diff
@@ -4,12 +4,12 @@
 def Sign_Extender(clk, instruction, output):
     #Instruction should be a 16-bit intbv
     #output is accurate 32-bit intbv.
-    output = intbv(max = 32)
+    output.Next = intbv(min = 32, max = 32)
```
Comment (mtvjr) [inline python/Sign_Extender.py]: It looks like the initial code was mistaken, using intbv(mine = 32, max = 32) would constrain the output to the numbers 32 or 32. One would need to do "... = intbv(min = 0, max = 2**32) to get the correct range.

File: python/Sign_Extender.py [hunk 2]
```diff
@@ -4,12 +4,12 @@
...
     
     @alway(clk.posedge)
     def execute():
         #Insert 16 to 32-bit conversion loginc
-
-        return
+        output[31:16].Next = instruciton[15] #make the first new bits match the first bit
```
Comment (mtvjr) [inline python/Sign_Extender.py]: Python doesn't allow arrays to be assigned a single value in batch. You might need to do a for loop ala
```
for out in output[16:31]:
  out.next = ..."
```

Also it looks like instruction is mispelled.

File: python/Sign_Extender.py [hunk 3]
```diff
@@ -4,12 +4,12 @@
...
+        return output.Next
```
Comment (mtvjr) [inline python/Sign_Extender.py]: Return values in the inner myhdl functions are useless. It's impossible to retrieve the value from them due to the round about way the function is called. Output is done through the passed output parameter which is already implemented in the file!

**Comment by MMaluschka13** [comment]
I updated the pull request. So I believe they should be adequate now.

**Pull request merged**